frontend/src/components/molecules/AddTransaction.tsx

```diff
@@ -22,39 +74,50 @@ const AddTransaction: React.FC<{ className?: string }> = props => {
     interval_type,
     interval,
   }: any) => {
-    if (!recurring) {
-      await TransactionActions.doCreateTransaction(
-        {
-          company_id: auth!.selectedCompany!,
-          date,
-          description,
-          money: money * 100,
-          notes,
-          type,
-        },
-        dispatch
-      );
-    } else {
-      await TransactionActions.doCreateRecurringTransaction(
-        {
-          company_id: auth!.selectedCompany!,
-          end_date,
-          interval,
-          interval_type,
-          start_date: date,
-          template: {
+    if (!(description.length > 140) && !(notes.length > 140)) {
+      state.content = 'Transaction added successfully';
+      state.clicker = onButtonClickHandler;
```

Comment:
Never mutate state in react! All updates to state should be done with `dispatch` or `setState`.